<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScanPathDiskEdit</class>
 <widget class="QWidget" name="ScanPathDiskEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>216</width>
    <height>378</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>圆盘扫描计划</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="0" column="0">
    <widget class="QFrame" name="parameterContainer">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="2" column="0">
       <widget class="QLabel" name="channelTitle">
        <property name="text">
         <string>通道</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="channel"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="scanAxisTitle">
        <property name="text">
         <string>轴</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="AxisSelector" name="scanAxis"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="intervalTitle">
        <property name="text">
         <string>间隔</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="interval">
        <property name="suffix">
         <string> mm</string>
        </property>
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>9999.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
        <item>
         <widget class="QLabel" name="suggestedSpeed">
          <property name="text">
           <string>最大角速度：0 °/s</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="useSuggestedSpeed">
          <property name="text">
           <string>使用推荐速度</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="angularSpeedTitle">
        <property name="text">
         <string>扫描速度</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QDoubleSpinBox" name="angularSpeed">
        <property name="suffix">
         <string> °/s</string>
        </property>
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>9999.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="coordinateSystemTitle">
        <property name="text">
         <string>坐标系</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QComboBox" name="coordinateSystem"/>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="originTitle">
        <property name="text">
         <string>原点</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QLabel" name="origin">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QCheckBox" name="axisRotationEnable">
        <property name="text">
         <string>有限旋转轴</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="AxisSelector" name="axesRotationLimited">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="setupEnable">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>使用此工艺</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="setup">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>编辑...</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="endPointLabel">
        <property name="text">
         <string>机械臂</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="EndPointSelector" name="endPointUid"/>
      </item>
      <item row="10" column="0">
       <widget class="QCheckBox" name="rectangleEnable">
        <property name="text">
         <string>矩形成像</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupStart">
     <property name="title">
      <string>起点</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="PointXYZ" name="startPoint" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="groupEnd">
     <property name="title">
      <string>终点</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="PointXYZ" name="endPoint" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EndPointSelector</class>
   <extends>QComboBox</extends>
   <header>Motion/EndPointSelector.h</header>
  </customwidget>
  <customwidget>
   <class>AxisSelector</class>
   <extends>QComboBox</extends>
   <header>Acquisition/AxisSelector.h</header>
  </customwidget>
  <customwidget>
   <class>PointXYZ</class>
   <extends>QWidget</extends>
   <header>Acquisition/PointXYZ.h</header>
   <container>1</container>
   <slots>
    <signal>pointChanged()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>setupEnable</sender>
   <signal>toggled(bool)</signal>
   <receiver>setup</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>36</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>107</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>axisRotationEnable</sender>
   <signal>toggled(bool)</signal>
   <receiver>axesRotationLimited</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel">
     <x>104</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>interval</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ScanPathDiskEdit</receiver>
   <slot>refreshSpeed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>205</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refreshSpeed()</slot>
 </slots>
</ui>
